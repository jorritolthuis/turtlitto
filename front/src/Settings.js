import React from "react";
import PropTypes from "prop-types";
import Turtle from "./Turtle";
import TurtleEnableBar from "./TurtleEnableBar";

/**
 * Show the settings of all turtles.
 * Author: H.E. van der Laan
 *
 * props:
 * - turtles: an array of Turtles
 * - onTurtleEnableChange: function to call when the turtle enable button is pressed
 */
const Settings = props => {
  const { turtles, onTurtleEnableChange } = props;
  return (
    <div>
      <TurtleEnableBar
        turtles={turtles.map(turtle => {
          return {
            id: turtle.id,
            enabled: turtle.enabled
          };
        })}
        onTurtleEnableChange={onTurtleEnableChange}
      />
      {turtles
        .filter(turtle => turtle.enabled)
        .map(turtle => (
          <Turtle
            key={turtle.id}
            turtle={turtle}
            editable
            onChange={(changedProp, newValue) => {} /* TODO: turtle update */}
          />
        ))}
    </div>
  );
};

Settings.propTypes = {
  onTurtleEnableChange: PropTypes.func.isRequired,
  turtles: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      enabled: PropTypes.bool.isRequired,
      battery: PropTypes.number.isRequired,
      home: PropTypes.string.isRequired,
      role: PropTypes.string.isRequired,
      team: PropTypes.string.isRequired
    })
  ).isRequired
};

export default Settings;
