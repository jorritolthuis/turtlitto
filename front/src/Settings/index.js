import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Dropdown from "../Dropdown";
import TurtleList from "../TurtleList";
import sendToServer from "../sendToServer";

/**
 * Show the settings of all turtles.
 * Author: H.E. van der Laan
 * Author: T.T.P. Franken
 *
 * props:
 * - turtles: an array of Turtles
 * - session: a string which holds the password needed to connect to the SRRS
 */

const CONFIG_VALUES = [
  "Role assigner on",
  "Role assigner off",
  "Pass demo",
  "Penalty mode",
  "Ball Handling demo"
];

const COMMAND_VALUES = {
  "Role assigner on": "role_assigner_on",
  "Role assigner off": "role_assigner_off",
  "Pass demo": "pass_demo",
  "Penalty mode": "penalty_demo",
  "Ball Handling demo": "ball_handling_demo"
};

const Settings = props => {
  const { turtles, session } = props;
  return (
    <SettingsWrapper>
      <RoleDropdown
        id={"settings_role-dropdown"}
        currentValue={"Whatever"}
        values={CONFIG_VALUES}
        onChange={value => {
          sendToServer(COMMAND_VALUES[value], "command", session);
        }}
        enabled={true}
      />
      <TurtleList turtles={turtles} session={session} />
    </SettingsWrapper>
  );
};

Settings.propTypes = {
  turtles: PropTypes.objectOf(
    PropTypes.shape({
      enabled: PropTypes.bool.isRequired,
      batteryvoltage: PropTypes.number.isRequired,
      homegoal: PropTypes.string.isRequired,
      role: PropTypes.string.isRequired,
      teamcolor: PropTypes.string.isRequired
    })
  ).isRequired
};

const RoleDropdown = styled(Dropdown)`
  margin-left: auto;
  margin-right: auto;
  height: 4rem;
`;

const SettingsWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

export default Settings;
