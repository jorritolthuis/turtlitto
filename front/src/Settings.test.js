import React from "react";
import { shallow } from "enzyme";
import Settings from "./Settings";

it("renders without crashing", () => {
  const wrapper = shallow(
    <Settings
      turtles={[
        {
          id: 1,
          enabled: true,
          battery: 66,
          home: "Yellow home",
          role: "INACTIVE",
          team: "Magenta"
        },
        {
          id: 2,
          enabled: false,
          battery: 42,
          home: "Yellow home",
          role: "INACTIVE",
          team: "Magenta"
        }
      ]}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
