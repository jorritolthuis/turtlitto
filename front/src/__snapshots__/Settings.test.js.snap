// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <TurtleEnableBar
    onTurtleEnableChange={[Function]}
    turtles={
      Array [
        Object {
          "enabled": true,
          "id": 1,
        },
        Object {
          "enabled": false,
          "id": 2,
        },
      ]
    }
  />
  <Turtle
    editable={true}
    key="1"
    onChange={[Function]}
    turtle={
      Object {
        "battery": 66,
        "enabled": true,
        "home": "Yellow home",
        "id": 1,
        "role": "INACTIVE",
        "team": "Magenta",
      }
    }
  />
</div>
`;
